scripts-components/shapes-fallback/src/index.ts

```diff
@@ -0,0 +1,69 @@
+import { Component } from '@mangoweb/scripts-base'
+
+export interface ShapesFallbackProps {}
+
+export function isShapesFallbackRequired() {
+	const isIE = navigator.userAgent.indexOf('MSIE ') > 0 || navigator.userAgent.indexOf('Trident/') > 0
+	return isIE
+}
+
+const loadedSymbolsLocations: string[] = []
+
+function loadSymbolsLocation(location: string) {
+	loadedSymbolsLocations.push(location)
+
+	const xhr = new XMLHttpRequest()
+	xhr.open('GET', location)
+	xhr.onload = function() {
+		if (xhr.status === 200) {
+			const wrapper = document.createElement('div')
+			const body = document.body
+
+			wrapper.innerHTML = xhr.response
+
+			const el = wrapper.children.item(0)
+
+			if (el) {
+				body.insertBefore(el, body.firstChild)
+			}
+		}
+	}
+	xhr.send()
+}
+
+export function applyShapesFallbackIfNeeded(parent: HTMLElement) {
+	if (!isShapesFallbackRequired()) {
+		return
+	}
+
+	const shapes = parent.querySelectorAll('svg use')
+	for (let i = 0; i < shapes.length; i++) {
+		const shape = shapes[i]
+		if (!shape.hasAttribute('xlink:href')) {
+			continue
+		}
+
+		const href = shape.getAttribute('xlink:href') || ''
+		const hrefParts = href.split('#')
+		if (hrefParts.length < 2) {
+			continue
+		}
+
+		const symbolsLocation = hrefParts[0]
+		const symbolId = hrefParts[1]
+
+		shape.setAttribute('xlink:href', `#${symbolId}`)
+
+		if (loadedSymbolsLocations.indexOf(symbolsLocation) < 0) {
```

Comment:
To pole vůbec nemusí být v module scope, ale může být nad touhle smyčkou.

Reply:
Určitě? V module scope to řeší více volání `applyShapesFallbackIfNeeded`. Třeba situaci, kdy při načtení stránky máme v html `shapes-1.svg` a `shapes-2.svg` a později ajaxem donačteme html, ve kterém je znovu `shapes-1.svg`. Mít pole `loadedSymbolsLocations` až nad for smyčkou by vedlo k tomu, že `shapes-1.svg` se do stránky injectne dvakrát, ne?

Reply:
Hezky se to zarovnalo. :D

![image](https://user-images.githubusercontent.com/1045362/81059447-b89f2280-8ed0-11ea-9039-92d1cbeb2a25.png)